### ISSUE TITLE
Handling mfa_required response?

### ISSUE THREAD

**Issue by ledjon** [auth0/auth0-java#119]
According to the [auth0 docs](https://auth0.com/docs/api-auth/tutorials/multifactor-resource-owner-password#prerequisites) we can use the auth api to login users that have MFA enabled.

The behavior is that the API will respond (after checking user/pass) with a 403 and the body of something like

```
{
  "error": "mfa_required",
  "error_description": "Multifactor authentication required",
  "mfa_token": "eyJ0eXAiOiJKV1QiLCJhbGci....D3QCiQ"
}
```

You can then take the mfa_token and continue the response cycle with your end user.

The way this sdk is written write now all the details about the error are lost except the "error" key.  The mfa_token is effectively discarded when it is passed to the APIException exception to be thrown.

Is there something I am missing here? is there any place to address this flow? Am I doing something wrong with how we would handle this using the auth api and this sdk?

Thanks.

**Comment by ledjon** [comment]
Thank you for the quick response.  I went on reading the docs and looking at what the SDK could do and discovered that there are probably a couple of gaps that would need to be addressed.  But yes reading that property from the error response would be a great first step, as that would allow me to get up to that point using the SDK and then only have to call a limited number of APIs directly.

**Comment by ledjon** [comment]
@lbalmaceda thank you very much for getting this change implemented and merged in.

From what I could tell of the change this allows the caller to get accss to the extra fields in the error which is a great first step.

Is there a plan to implement the rest of the mfa API calls needed to do the full workflow via the sdk?   I don't remember the specifics right now but I'm pretty sure there are some extra APIs that need to be called with the challenge/response to the mfa that weren't supported in the sdk.

Thanks a million!

**Comment by lbalmaceda** [comment]
Correct, the introduced changes let you obtain any value from the response error object. Regarding the missing MFA endpoints I have added that to our TODO list but I can't give you an ETA. You should be able to call them yourself, maybe the `CustomRequest` class helps you to construct them. Eventually all endpoints both for authentication and management will get implemented. 🎉
